src/ui/public/vislib/vis.js

```diff
@@ -47,15 +47,7 @@ export function VislibVisProvider(Private) {
 
       this.data = data;
 
-      if (!this.uiState) {
```

Comment:
vislib did listen on changes in the `uiState` earlier. This isn't needed, since `visualize` (if used) or otherwise `visualization` will listen on changes to the `uiState` anyway and trigger a rerender. This were also the cause for occasional multiple render steps when uiState changed.